src/Controls/GeosearchOffMap.js

```diff
@@ -0,0 +1,292 @@
+    }
+
+    this._geosearchClass.addEventParent(this);
+
+    this._geosearchClass._pendingSuggestions = [];
+
+    this.setup(options);
+  },
+
+  _renderSuggestions: function (suggestions) {
+    var currentGroup;
+    this._suggestions.style.display = 'block';
+
+    var nodes = [];
+    var list;
+    var header;
+
+    for (var i = 0; i < suggestions.length; i++) {
+      var suggestion = suggestions[i];
+      if (!header && this._geosearchClass._providers.length > 1 && currentGroup !== suggestion.provider.options.label) {
+        header = L.DomUtil.create('span', 'geocoder-control-header', this._suggestions);
+        header.textContent = suggestion.provider.options.label;
+        header.innerText = suggestion.provider.options.label;
+        currentGroup = suggestion.provider.options.label;
+        nodes.push(header);
+      }
+
+      if (!list) {
+        list = L.DomUtil.create('ul', 'dropdown-menu', this._suggestions);
+        list.style.display = 'block';
+      }
+      var suggestionItem = L.DomUtil.create('li', 'anything', list);
+      
+      // need to wrap in an <a> tag to get selection CSS and good spacing
+      suggestionItem.innerHTML = '<a>' + suggestion.text + '</a>';
+      
+      // which makes it necessary to nest the provider and magicKey one step deeper
+      // this code is hideous
+      suggestionItem.childNodes[0].provider = suggestion.provider;
+      suggestionItem.childNodes[0]['data-magic-key'] = suggestion.magicKey;
+    }
+
+    nodes.push(list);
+
+    return nodes;
+  },
+
+  clear: function () {
+    this._suggestions.innerHTML = '';
+    this._suggestions.style.display = 'none';
+    // L.DomUtil.addClass(this._suggestions, 'dropdown-menu');
+
+    this._input.value = '';
+
+    if (this.options.collapseAfterResult) {
+      this._input.placeholder = '';
+      // L.DomUtil.removeClass(this._wrapper, 'geocoder-control-expanded');
+    }
+
+    // if (!this._map.scrollWheelZoom.enabled() && this._map.options.scrollWheelZoom) {
+    //   this._map.scrollWheelZoom.enable();
+    // }
+  },
+
+  clearSuggestions: function () {
+    if (this._nodes) {
+      for (var k = 0; k < this._nodes.length; k++) {
+        if (this._nodes[k].parentElement) {
+          this._suggestions.removeChild(this._nodes[k]);
+        }
+      }
+    }
+  },
+
+  getAttribution: function () {
+    var attribution = this.options.attribution;
+
+    for (var i = 0; i < this._providers.length; i++) {
+      attribution += (' ' + this._providers[i].options.attribution);
+    }
+
+    return attribution;
+  },
+
+  setup: function (options) {
+    this._map = options.map;
+    this._input = document.getElementById(options.inputTag)
+    L.DomUtil.addClass(this._input, 'the-input');
+
+    // more neckbeardy if i can stop using jquery
+    this._wrapper = $( ".the-input" ).wrap(function() {
+      return "<div id= 'address-foobaz' class='input-group' style='width:100%'></div>";
```

Comment:
`address-foobaz` as the ID is a little odd. I'm assuming you wanted a unique id? The best way to do that is to keep a private counter of how many times this has been called and generate an ID with that number.